students/adam_strong/lesson05/mailroom3.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+# Mailroom2.py - Lesson04 Assignment
+import sys
+import pathlib
+
+# Changed the list into a dict
+donor_db = {"Scrooge McDuck": [8000.00, 70000.00],
+            "Montgomery Burns": [49.53],
+            "Richie Rich": [1000000.00, 500000.00],
+            "Chet Worthington": [200.00, 44387.63, 10200.00],
+            "Silas Skinflint": [0.25, 1.00, 0.43]}
+
+main_prompt = "\n".join(("", "Welcome to the donors list",
+          "Please choose from below options:",
+          "1 - Send a thank you",
+          "2 - Create a report",
+          "3 - Send a letter to all the donors",
+          "4 - Quit",
+          "Type a number to select >>> "))
+
+ty_prompt = "\n".join(("", "Please type the full name of the donor OR",
+        "type 'list' to see a list of donors",
+        "Type input here >>>"))
+
+ty_message = "\n".join(("", "Dear {}",
+        "Thank you for your generous donation of {:.2f}",))
+
+letter = "\n".join(("", "Dear {},","",
+        "    Thank you for your very kind donations totaling ${:.2f}.","",
+        "    It will be put to very good use.","",
+        "               Sincerely,",
+        "                  - The team"))
+
+def exit_program():
+    print('\nShutting down the program\n')
+    sys.exit()
+
+def thank_you():
+    tyname = input(ty_prompt)
+    if tyname == 'list':
+        report()
+    try:
+        amt = input("Please enter the donation amount >>>")
+        if donor_db.get(tyname) == None:
+            donor_db[tyname] = [float(amt)]
+        else:
+            donor_db.get(tyname).append(float(amt))
+        input(ty_message.format(tyname, float(amt)))
+        print('')
+    except ValueError:
```

Comment:
a good rule of thumb is to try to dial in exception handling to the code that would throw that exception.

so for example you could do:
`amt = float(mat)` and put the exception handling just around that line